var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -545,7 +545,10 @@ def install(self, spec, prefix):
 
         threading_opts = self.determine_b2_options(spec, b2_options)
 
-        b2('--clean')
+        if spec.satisfies('@:1.65'):
+            b2('--clean', *b2_options)
+        else:
+            b2('--clean')
```

Comment:
```suggestion
        b2('--clean', *b2_options)
```

How about we just pass all b2 options to b2 all the time? That simplifies things a bit, right?

Reply:
ping @hainest

Reply:
Also, the same happens to boost 1.66

Reply:
`b2 --instal`l option is fixed.

When icu is cached, the header file of icu is used.
If you specify  `spack install boost@1.65 ~icu cxxstd=98` , the build is failed.